x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/nio/SecurityNioHttpServerTransport.java

```diff
@@ -91,6 +89,7 @@ public NioHttpChannel createChannel(NioSelector selector, SocketChannel channel)
             NioHttpChannel httpChannel = new NioHttpChannel(channel);
             HttpReadWriteHandler httpHandler = new HttpReadWriteHandler(httpChannel,SecurityNioHttpServerTransport.this,
                 handlingSettings, corsConfig, selector.getTaskScheduler(), threadPool::relativeTimeInNanos);
+            NioIPFilter handler = new NioIPFilter(httpHandler, httpChannel.getRemoteAddress(), ipFilter, IPFilter.HTTP_PROFILE_NAME);
```

Comment:
Can't we just not add this handler if we don't have any IP filtering? I.e. do this:

```java
            final NioChannelHandler handler;
            if (ipFilter != null) {
                handler = new NioIPFilter(httpHandler, httpChannel.getRemoteAddress(), ipFilter, IPFilter.HTTP_PROFILE_NAME);
            } else {
                handler = httpHandler;
            }
```

the we wouldn't have to bother with the whole `null` check business in the `NioIPFilter` and save a bit of indirection?